app/renderer/components/preferences/syncTab.js

```diff
@@ -27,14 +27,25 @@ class SyncTab extends ImmutableComponent {
     this.enableRestore = this.enableRestore.bind(this)
   }
 
+  get setupError () {
+    return this.props.syncData.get('setupError')
+  }
+
   get isSetup () {
-    return this.props.syncData.get('seed') instanceof Immutable.List && this.props.syncData.get('seed').size === 32
+    return !this.setupError && this.props.syncData.get('seed') instanceof Immutable.List && this.props.syncData.get('seed').size === 32
   }
 
   get enabled () {
     return getSetting(settings.SYNC_ENABLED, this.props.settings)
   }
 
+  get errorContent () {
+    return <div className='errorContainer'>
+      <div className='setupError'>{this.setupError}</div>
+      <button data-l10n-id='syncRetryButton' onClick={this.retry.bind(this)} />
```

Comment:
this looks like a little basic; what do you think of: 
`<Button l10nId='syncRetryButton' className='primaryButton' onClick={this.retry.bind(this)} />`